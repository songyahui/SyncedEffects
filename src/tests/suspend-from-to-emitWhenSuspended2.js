"use hopscript"

const hh = require("hiphop");

const m = new hh.ReactiveMachine(
   <hh.module S R E>
     <hh.loop>
       <hh.suspend immediate from=S to=R emitWhenSuspended=E>
	 <hh.atom apply=${() => console.log("not suspended!")}/>
       </hh.suspend>
       <hh.pause/>
     </hh.loop>
   </hh.module>
);

m.debug_emitted_func = emitted => {
   console.log(emitted);
   console.log("---------------------");
};
// m.debuggerOn("debug");
// m.stepperOn();
m.react()
m.react()
m.inputAndReact("S");
m.react()
m.react()
m.inputAndReact("R");
m.react()
m.react()
