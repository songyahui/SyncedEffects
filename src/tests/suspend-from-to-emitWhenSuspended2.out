not suspended!
[]
---------------------
not suspended!
[]
---------------------
[ 'S', 'E' ]
---------------------
[ 'E' ]
---------------------
[ 'E' ]
---------------------
not suspended!
[ 'R' ]
---------------------
not suspended!
[]
---------------------
not suspended!
[]
---------------------
